lib/matplotlib/testing/compare.py

```diff
@@ -146,16 +146,15 @@ def __call__(self, orig, dest):
             weakref.finalize(self._tmpdir, self.__del__)
         if (not self._proc  # First run.
                 or self._proc.poll() is not None):  # Inkscape terminated.
+            if self._proc is not None and self._proc.poll() is not None:
+                for stream in filter(None, [self._proc.stdin,
+                                            self._proc.stdout,
+                                            self._proc.stderr]):
+                    stream.close()
             env = {
                 **os.environ,
-                # If one passes e.g. a png file to Inkscape, it will try to
-                # query the user for conversion options via a GUI (even with
-                # `--without-gui`).  Unsetting `DISPLAY` prevents this (and
-                # causes GTK to crash and Inkscape to terminate, but that'll
-                # just be reported as a regular exception below).
-                "DISPLAY": "",
```

Comment:
Making INSCAPE_PROFILE_DIR a real directory seems like a good idea, but are you sure you can keep DISPLAY being set (assuming an X11 setup) and not have the interactive conversion options window pop up?  At least for some older version of inkscape that was, I believe, not the case.  (Or we could just document we need Inkscape > (some version), which is fine too.)

Reply:
... and now it is working.  Dropping this commit.